ph5/clients/ph5toms.py

```diff
@@ -602,6 +602,13 @@ def create_trace(self, station_to_cut, mp=False):
                         self.ph5.get_receiver_t_by_n_i(stc.receiver_n_i)
                     azimuth = Receiver_t['orientation/azimuth/value_f']
                     dip = Receiver_t['orientation/dip/value_f']
+                    # dip is below horizontal axis
+                    # but SAC convention requires from vertical axis.
+                    dip = float(dip)
+                    if Receiver_t['orientation/description_s'] == 'Z':
```

Comment:
Why does the Z component control if the dip is 90 +dip vs 90-dip? Should all components be 90+dip or 90-dip? 

The SAC documentation says: 
"Component incident angle (degrees from upward vertical; SEED/MINISEED uses dip: degrees from horizontal down)."

Sorry for requesting clarification.

Reply:
I didn't look at the document of SAC, just follow the request from Holly which is stated above https://github.com/PIC-IRIS/PH5/pull/496#issuecomment-1018848379.
@hrotman-pic Holly, can you please help us to understand more about this? Thank you.

Reply:
I'm confused about the logic as well, but I am maybe missing something.

Reply:
If all components were 90-dip, CMPINC looks like this for a Fairfield SEG-D:

SAC> lh CMPINC


  ###### FILE: XE.221..DP1.2020-10-27T01.00.00.sac - 1
 -----------------------------------------

     CMPINC = 9.000000e+01

  ###### FILE: XE.221..DP2.2020-10-27T01.00.00.sac - 2
 -----------------------------------------

     CMPINC = 9.000000e+01

  ###### FILE: XE.221..DPZ.2020-10-27T01.00.00.sac - 3
 -----------------------------------------

     CMPINC = 0.000000e+00


CMPINC is dip+90 for Z channel on the version I approved so CMPINC will match the FDSNWS SAC examples (am assuming FDSNWS dataselect SAC is a reliable source), which is why I provided those examples.  Most PH5s experiments should look like the Z9.M01 FDSNWS example.  Rarely, one should look like the 2B.PETZ FDSNWS example.

@ascire-pic can elaborate, confirm, or deny if needed.

Reply:
Or, referring back to the SEED dip using a subset of the experiments I checked for testing & comparison.

20-001, XE.221.DPZ, SEED dip=90 (reversed convention): http://ds.iris.edu/mda/XE/221/--/DPZ/?starttime=2020-10-24&endtime=2020-11-25

SEED station Z9.M01.DHZ SEED dip=90 (reversed convention): http://ds.iris.edu/mda/Z9/M01/--/DHZ/?starttime=2019-01-04&endtime=2020-06-12

SEED dip is the same for these, so SAC CMPINC should be the same for these, and they are the same (CMPINC=180) on the current version.


16-015, YW.502.HHZ, SEED dip=-90: http://ds.iris.edu/mda/YW/502/01/HHZ/?starttime=2016-06-20&endtime=2016-10-30

SEED station 2B.PETZ.HHZ, SEED dip=-90: http://ds.iris.edu/mda/2B/PETZ/--/HHZ/?starttime=2019-10-21&endtime=2021-11-03

SEED dip is the same for these, so SAC CMPINC should also be the same for these, and they are (CMPINC=0) on the current version as well.

Does that also help?

Reply:
I think that the logic above is almost right but needs to have the signs flipped so :

PNCINC_Z = 90 - dip_Z
and 
PNCINC_H = 90 + dip_H

All of the components in the instrument are tied together at 90 degrees. If we have vertical components, PNCINC_Z = 90 - dip_Z, and horizontal components PNCINC_H = 90 + dip_H we'll keep this 90 degree relationship and we'll follow the `degrees from upward vertical` == clockwise convention. 

An example with the current logic described immediately above: 
 dip_Z = 85 
dip_H = 5 
meaning 
PNCINC_Z=5 
PNCINC_H = 95

What does everyone think?

<img width="378" alt="Screen Shot 2022-01-28 at 10 56 06 AM" src="https://user-images.githubusercontent.com/40644784/151605259-45dcc20d-c029-49ff-803d-227ca96d0f76.png">

Reply:
This logic suggests that the CMPINC header is being written incorrectly on FDSN web service queries.

Reply:
Can you provide an example? Are we using a counter clockwise convention? I am, okay with whatever is decided.

Reply:
I have examples in my approving review, https://github.com/PIC-IRIS/PH5/pull/496#pullrequestreview-865103223, and link to the SEED dip for the one example of each in https://github.com/PIC-IRIS/PH5/pull/496#discussion_r794562682.  Please let me know if those don't make sense.  I realize my review is very long.

Reply:
-180 and 180 are the same position on the unit circle. The question is if we want our convention to be clockwise or counter clockwise.

Reply:
@hrotman-pic you are right, and I am off by a negative sign. I think we should merge the current logic.